### PR TITLE
How do I add a new variable to dataset.

### LINKED ISSUE #3678: Feature request: How Do I add a new variable to a data set?
A common thing to want to do is to create a new `DataArray` and add it to an existing `Dataset`, but the answer does not seem to appear in the "How Do I?" list.

Thanks!

### PR DESCRIPTION
**Pull request by rpgoldman**
Closes #3678 

I don't have the full build environment: someone who does will have to make sure that I got the sphinx formatting right.

### CODE REVIEW

File: doc/howdoi.rst [hunk 1]
```diff
@@ -11,6 +11,8 @@ How do I ...
 
    * - How do I...
      - Solution
+   * - add a DataArray to my dataset as a new variable
+     - ``my_dataset[varname] = my_dataArray``
```
Comment (dcherian) [inline doc/howdoi.rst]:
```suggestion
     - ``my_dataset[varname] = my_dataArray`` or :py:meth:`Dataset.assign`
```

We should also link to https://xarray.pydata.org/en/stable/data-structures.html#dictionary-like-methods. We have docs on building the documentation here: https://xarray.pydata.org/en/stable/contributing.html#contributing-to-the-documentation

**Comment by rpgoldman** [comment]
OK, I have tried to do as you suggested.

I looked at the directions for building the docs as you suggested, and I'm afraid it's too big an ask for me.  I don't use conda, and I can't spare the time to learn it.  Similarly, I know enough RST to write numpy-style docstrings, but I don't have the time to learn Sphinx.  With all due respect, if xarray wants contributions from people like me, something has to be done to make this easier.

I'm just doing this because I'm contributing to Arviz, which depends on xarray.  I can try to fix things if I see something wrong in the "Docs" check, but that's all I can do, I'm afraid.

**Comment by rpgoldman** [comment]
I didn't get the cross-reference syntax right. I've tried to fix it, and am waiting for a new check.

**Comment by rpgoldman** [comment]
OK, I'm not trying to be difficult, but I think someone else who knows how RST works could fix this in 2.4 seconds, or I could bang my head on it for hours.

Would someone please just fix my syntax error and merge this?

File: doc/howdoi.rst [hunk 2]
```diff
@@ -11,6 +11,8 @@ How do I ...
 
    * - How do I...
      - Solution
+   * - add a DataArray to my dataset as a new variable
+     - ``my_dataset[varname] = my_dataArray`` or :py:meth:`Dataset.assign`[#assign_dict]_
```
Comment (keewis) [inline doc/howdoi.rst]: I think you could use
```suggestion
     - ``my_dataset[varname] = my_dataArray`` or :py:meth:`Dataset.assign` (see also :ref:`dictionary_like_methods`)
```
then you wouldn't need the footnote.

**Comment by dcherian** [comment]
Great. Thanks @rpgoldman Welcome to xarray!

**Pull request merged**